website/public/js/controllers/settingsCtrl.js

```diff
@@ -66,12 +66,23 @@ habitrpg.controller('SettingsCtrl',
       User.set({'flags.newStuff':true});
     }
 
-    $scope.saveDayStart = function(){
-      var dayStart = +User.user.preferences.dayStart;
-      if (_.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
+    $scope.passDayStart = User.user.preferences.dayStart;
+
+    $scope.saveDayStart = function(newDayStart){
+      var oldDayStart = User.user.preferences.dayStart;
+      var dayStart = newDayStart;
+      var lastCron = User.user.lastCron;
+      var getOldStart = Shared.startOfDay({ dayStart: oldDayStart});
+      var getNewStart = Shared.startOfDay({ dayStart: dayStart});
+      var isoNewStart = Shared.isoTimestamp(getNewStart);
+
+      if (dayStart == undefined || _.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
```

Comment:
The only thing I don't like about `!dayStart` is it returns true for zero (Test case 1).

I did a bunch of tests. If I apply `Math.floor` right away in the var statement, `_.isNaN(dayStart)` works quite nicely, but the empty case, which was returning `null` in my previous tests, suddenly decided it was really a zero, and I don't think we want to just set someone's dayStart offset to zero if they happen leave the control after deleting the existing value. I was leaning toward `(!dayStart && dayStart != 0)`, but I also didn't like that `23.5` doesn't behave the same as `22.5` because the control passes anything greater than `23` as undefined.

On a lark, I tested `dayStart != Math.floor(dayStart)`, and I think that might be a good solution. It returns true for any values in `-1, 5.5, k, 23.5, 24, 25, null`. It returns false for all values in `0, 1, 5, 23`. This means we expect people to understand 5.5 is an invalid value when they get the error message, which really is implied by `_____:00 (24 hr clock)` on the page, but not so clear in the error message.

So if you agree, I'll use `if ( dayStart != Math.floor(dayStart) || dayStart < 0 || dayStart > 24 ) {error message}`.